src/components/Layer.js

```diff
@@ -1,756 +1,461 @@
+      </Accordion.Toggle>
+
+      <Accordion.Collapse eventKey={state.ukey} >
+
+        <Card.Body>
+
+          <Form action="" >
+
+            <Form.Group controlId="formLayerName">
+              <Form.Label>Name</Form.Label>
+              <Form.Control
+                type="text"
+                className="form-control"
+                placeholder="Layer name"
+                defaultValue={state.name}
+                onChange={handleNameChange}
+                name="name"
+              />
+            </Form.Group>
+
+
+            <LayerTypeForm
+              state={state}
+              handleLayerTypeChange={handleLayerTypeChange}
+            />
+
+            {
+              state.layerType === LAYER_TYPE_LATLON && (
+                <LatLonLayerForm
+                  state={state}
+                  handleNodeLabelChange={handleNodeLabelChange}
+                  handleLatPropertyChange={handleLatPropertyChange}
+                  handleLonPropertyChange={handleLonPropertyChange}
+                  handleTooltipPropertyChange={handleTooltipPropertyChange}
+                  handleLimitChange={handleLimitChange}
+                />
+              )
+            }
+            {
+              state.layerType === LAYER_TYPE_POINT && (
+                <PointLayerForm
+                  state={state}
+                  handleNodeLabelChange={handleNodeLabelChange}
+                  handleTooltipPropertyChange={handleTooltipPropertyChange}
+                  handlePointPropertyChange={handlePointPropertyChange}
+                  handleLimitChange={handleLimitChange}
+                />
+              )
+            }
+            {
+              state.layerType === LAYER_TYPE_SPATIAL && (
+                <SpatialLayerForm
+                  state={state}
+                  handleTooltipPropertyChange={handleTooltipPropertyChange}
+                  handleSpatialLayerChanged={handleSpatialLayerChanged}
+                />
+              )
+            }
+            {
+              state.layerType === LAYER_TYPE_CYPHER && (
+                <CypherLayerForm
+                  state={state}
+                  handleCypherChange={handleCypherChange}
+                />
+              )
+            }
+
+            <MapRenderingForm
+              state={state}
+              handleRenderingChange={handleRenderingChange}
+              handleColorChange={handleColorChange}
+              handleRadiusChange={handleRadiusChange}
+            />
+
+            {state.ukey !== NEW_LAYER.ukey && (
+              <Button variant="danger" type="submit" onClick={handleDeleteLayer} hidden={layer === undefined}>
+                Delete Layer
+              </Button>
+            )}
+
+            <Button variant="info" type="submit" onClick={handleShowQuery} hidden={state.layerType === LAYER_TYPE_CYPHER}>
+              Show query
+						</Button>
+
+            <Button variant="success" type="submit" onClick={handleUpdateLayer} >
```

Comment:
I know it was from a previous PR, but realized afterwards so just putting the comment here:

I think it should be to the user to decide whether he wants to create or update a layer, the app should know whether the layer already exists or not.

Reply:
And actually, the addLayer and updateLayer functions could be merged (we don't have to worry about performances, we'll never have hundreds of layers).

Reply:
I was more concerned with single responsibility for each declared function rather than perf :)

Reply:
> I think it should be to the user to decide whether he wants to create or update a layer, the app should know whether the layer already exists or not.

Exactly. Having two buttons explicitly require a user to make the decision if they want to update or create new. Functionally, that decision also means a user can quickly clone a new layer out of an old one if that's desirable